cmd/ipfs/daemon.go

```diff
@@ -461,3 +482,80 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
+	disableGC, _, err := req.Option(disableGCKwd).Bool()
+	if err != nil {
+		return err
+	}
+	if disableGC {
+		return nil
+	}
+
+	cfg, err := req.InvocContext().GetConfig()
+	if err != nil {
+		return fmt.Errorf("runGC: GetConfig() failed: %s", err)
+	}
+	ctx := req.Context()
+
+	// check if cfg has these fields initialized
+	// TODO: there should be a general check for all of the cfg fields
+	// maybe distinguish between user config file and default struct?
```

Comment:
this is a good idea. im not yet sure what's best way of handling this-- there's lots of complexity in how config files change.